PSReadLine/ViRegister.cs

```diff
@@ -0,0 +1,239 @@
+﻿using System;
+using System.Text;
+
+namespace Microsoft.PowerShell
+{
+    public partial class PSConsoleReadLine
+    {
+        /// <summary>
+        /// Used to report when the buffer is about to change
+        /// as part of a paste operation.
+        /// </summary>
+        public sealed class PasteEventArgs : EventArgs
+        {
+            /// <summary>
+            /// The text being pasted.
+            /// </summary>
+            public string Text { get; set; }
+            /// <summary>
+            /// The position in the buffer at
+            /// which the pasted text will be inserted.
+            /// </summary>
+            public int Position { get; set; }
+            /// <summary>
+            /// The recorded position in the buffer
+            /// from which the paste operation originates.
+            /// This is usually the same as Position, but
+            /// not always. For instance, when pasting a
+            /// linewise selection before the current line,
+            /// the Anchor is the cursor position, whereas
+            /// the Position is the beginning of the previous line.
+            /// </summary>
+            public int Anchor { get; set; }
+        }
+
+        /// <summary>
+        /// Represents a named register.
+        /// </summary>
+        public sealed class ViRegister

```

Comment:
Does this really need to be public? (I see it used in a test, but it becomes a supported API if it's public.)